tests/test_cross_mesh_resharding.py

```diff
@@ -242,14 +276,32 @@ def test_8gpu_2_dim_allgather(self):
             [NoSharding(), NoSharding(),
              NoSharding()], [Replicated(4)])
         self.run_resharding_task(src_shape, dst_shape, src_spec, dst_spec,
-                                  tensor_shape)
+                                  tensor_shape, resharding_mode="send_recv")
+        self.run_resharding_task(src_shape, dst_shape, src_spec, dst_spec,
+                                  tensor_shape, resharding_mode="broadcast")
+
+    def test_4gpu_broadcast(self):
+        src_shape = (1, 4)
+        dst_shape = (1, 4)
+        base = 64
+        tensor_shape = (2, 8*base, 8*base*8*base)
+
+        src_spec = ShardingSpec([Chunked([2]), Chunked([2]), NoSharding()], [ShardedAxis(0), ShardedAxis(1)])
+        dst_spec = ShardingSpec([NoSharding(), NoSharding(), NoSharding()], [Replicated(4)])
+        self.run_resharding_task(src_shape, dst_shape, src_spec, dst_spec,
+                                  tensor_shape, resharding_mode="broadcast")
 
+        src_spec = ShardingSpec([Chunked([2]), Chunked([2]), NoSharding()], [ShardedAxis(0), ShardedAxis(1)])
+        dst_spec = ShardingSpec([Chunked([2]), NoSharding(), Chunked([2])], [ShardedAxis(0), ShardedAxis(1)])
+        self.run_resharding_task(src_shape, dst_shape, src_spec, dst_spec,
+                                  tensor_shape, resharding_mode="broadcast")
 
 def suite():
     suite = unittest.TestSuite()
-    suite.addTest(ReshardingTest("test_4gpu_send_recv"))
+    suite.addTest(ReshardingTest("test_4gpu_all"))
     suite.addTest(ReshardingTest("test_4gpu_allgather"))
-    suite.addTest(ReshardingTest("test_8gpu_2_dim_allgather"))
+    suite.addTest(ReshardingTest("test_8gpu_2_dim_allgather_broadcast"))
+    suite.addTest(ReshardingTest("test_4gpu_broadcast"))
```

Comment:
The names and organizations of these unit tests are very confusing. I suggest the following modifications

1. keep all old test cases unchanged and use `send_recv` in the old test cases
2. add two new cases `test_4gpu_broadcast` and  `test_8gpu_broadcast`

Reply:
OK